src/pollers/statePoller.ts

```diff
@@ -0,0 +1,49 @@
+import { STATE_POLLING_INTERVAL } from "../globals/consts";
+import { getState } from "../binary/requests/requests";
+import { State } from "../binary/state";
+import { onStateChangedEmitter } from "../events/onStateChangedEmitter";
+
+let statePollingInterval: NodeJS.Timeout | null;
+let lastStateResponse: State | undefined | null;
+
+export async function pollState(): Promise<void> {
+  lastStateResponse = await getState();
+  if (lastStateResponse) {
+    onStateChangedEmitter.fire(lastStateResponse);
+  }
+
+  statePollingInterval = setInterval(
+    () => void doPollState(),
+    STATE_POLLING_INTERVAL
+  );
+}
+
+async function doPollState(): Promise<void> {
+  const state = await getState();
+  if (state && !equals(state, lastStateResponse)) {
+    lastStateResponse = state;
+    onStateChangedEmitter.fire(state);
+  }
+}
+
+export function cancelStatePolling(): void {
+  if (statePollingInterval) {
+    clearInterval(statePollingInterval);
+  }
+}
+
+function equals(state1: State, state2?: State | null): boolean {
```

Comment:
if you only care about the health check here so check this field only. there's no point in doing a "generic" equation here IMO

Reply:
actually no. I want every module to be able to use this poller

Reply:
but this way you're firing the event on unrelated changes. i would've let each consumer of the state decide whether or not the change is relevant to them

Reply:
I think it's ok. the consumer could check if it's relevant for it instead of repeatedly getting the state from the binary.

Reply:
we use the same mechanism in IJ
![image](https://user-images.githubusercontent.com/10290661/211417893-ae0cb1ab-2c66-4180-ab14-5fe952eaacc9.png)